tests/test_diagnostics_database.py

```diff
@@ -141,7 +141,7 @@ def test_example_cdd(self):
             'Vehicle_type': 'Sedan',
             'Special_setting': 3
         }
-        encoded_did = b'\x21\x03'
+        encoded_did = b'\x12\x03'
```

Comment:
I can't judge the patch itself, but this line seems to be _nibble_ order, not _byte_ order?

Reply:
This field position difference is a result of the different start-bit convention used by the codecs for fields that are declared big_endian vs little_endian regardless of their bit length. Because the current CDD loader mis-identifes the fields as little_endian the codecs are also misinterpreting the field start positions. (Compounded by the fact there is currently no mapping at all from the CDD start bit convention to the codec./DBC convention in the current CDD module)